### DISCUSSION TITLE
How to Build Routing Tiles for Multiple Countries

### DISCUSSION THREAD

**Discussion by plutus360** [valhalla/valhalla#4792] [Q&A] [answered]
Hello Valhalla team,
How can I build the routing tiles for multiple countries and run one application that can handle multiple countries?

example form : https://gis-ops.com/valhalla-part-2-how-to-run-valhalla-on-ubuntu/
```
valhalla_build_tiles -c ./conf/valhalla.json albania-latest.osm.pbf
find valhalla_tiles | sort -n | tar -cf "valhalla_tiles.tar" --no-recursion -T -
```

### run :  valhalla
```
valhalla_service ~/valhalla/scripts/conf/valhalla.json 2
```

**Comment by chrstnbwnkl** [comment, marked as answer]
Just merge the pbf files from different countries using your favorite merging tool (I use `osmium merge` for this, works like a charm)

**Reply by plutus360** [reply]
thank you I will try it !
